tests/unit/App.spec.js

```diff
@@ -269,4 +269,58 @@ Given("an App", () => {
       });
     });
   });
+
+  When("a response with link headers is returned", () => {
+    const rawLinks =
+      "</example>; rel=\"http://base.com\"; title=\"Example link\", " +
+      "</test>; rel=\"http://test.com\"; title=\"Test link\"";
+
+    beforeEach(() => {
+      app.setMethods({
+        request: app.vm.setLinks
+      });
+
+      app.vm.request(rawLinks);
+
+      app.update();
+    });
+
+    Then("one link should be displayed", () => {
+      expect(app.text().includes("Example link")).to.be.true;
+    });
+
+    Then("all the links should be displayed", () => {
+      expect(app.text().includes("Test link")).to.be.true;
+    });
+
+    Then("the link's href should be set correctly", () => {
+      expect(app.vm.links[0].href).to.equal("#http%3A%2F%2Fexample.com%2Fexample");
+    });
+
+    And("then a response without link headers is returned", () => {
```

Comment:
it's confusing that you say "then" at the beginning of all of these "And"s. Why don't you put these in different "When"s? it seems like you're repeating the "When" portion of this by doing app.vm.request again here.

Reply:
These tests are meant to check that the behavior is correct, *after* certain pre-conditions (in this case, correct link headers being returned). I prefer the `And then` syntax because it reads nicely, as `When X happens`, `And then Y happens`, `Then A should happen`. I would prefer not to move them to new `When` statements, because the setup would still have to be the same

Reply:
Looks good 👍